### PR TITLE
Removing Python 3.8

### PR DESCRIPTION
**Pull request by soma2000-lang**
#23044
@seberg  Dont know why the tests are still failing .Any pointers regarding the same would be  helpful

### CODE REVIEW

**Comment by seberg** [comment]
There are a lot of weird changes here, maybe easiest to start from main fresh and cherry-pick the good commits (before the merge) on top again?

**Comment by soma2000-lang** [comment]
@seberg ok.

File: doc/RELEASE_WALKTHROUGH.rst.txt [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
 ===================
 
+Add/drop Python versions
+------------------------
+
+When adding or dropping Python versions, three files need to be edited:
+
+- .github/workflows/wheels.yml  # for github cibuildwheel
+- .travis.yml  # for cibuildwheel aarch64 builds
+- setup.py  # for classifier and minimum version check.
```
Comment (mattip) [inline doc/RELEASE_WALKTHROUGH.rst.txt:33]: There are more CI runs in .github/workflows, as well as azure and cirrus. I am not sure that dropping/adding python versions belongs in the release walkthrough. We have some other CI documentation in the developer section.

File: doc/RELEASE_WALKTHROUGH.rst.txt [hunk 2]
```diff
@@ -1,36 +1,51 @@
...
+
 Backport Pull Requests
 ----------------------
 
 Changes that have been marked for this release must be backported to the
 maintenance/1.21.x branch.
 
-
-Update Release documentation
+Update release documentation
```
Comment (mattip) [inline doc/RELEASE_WALKTHROUGH.rst.txt:48]: Please try to limit this PR to things relevant for the issue at hand.

File: doc/neps/nep-0029-deprecation_policy.rst [hunk 3]
```diff
@@ -63,7 +63,7 @@ Consider the following timeline::
        |           |           |           |           |
   +++++|+++++++++++|+++++++++++|+++++++++++|+++++++++++|++++++++++++
    |              |                  |               |
-   py 3.5.0       py 3.6.0           py 3.7.0        py 3.8.0
+   py 3.5.0       py 3.6.0           py 3.7.0        py 3.9.0
```
Comment (mattip) [inline doc/neps/nep-0029-deprecation_policy.rst:66]: This is wrong. If at all, an additional date for py3.9.0 is needed.

File: doc/source/release/1.17.0-notes.rst [hunk 4]
```diff
@@ -7,7 +7,7 @@ NumPy 1.17.0 Release Notes
 This NumPy release contains a number of new features that should substantially
 improve its performance and usefulness, see Highlights below for a summary. The
 Python versions supported are 3.5-3.7, note that Python 2.7 has been dropped.
-Python 3.8b2 should work with the released source packages, but there are no
+Python 3.9b2 should work with the released source packages, but there are no
```
Comment (mattip) [inline doc/source/release/1.17.0-notes.rst:10]: No need to change this: the existing statement is correct for the 1.17 release

File: doc/source/release/1.17.3-notes.rst [hunk 5]
```diff
@@ -15,7 +15,7 @@ OpenBLAS >= 3.7 to avoid errors on the Skylake architecture.
 Highlights
 ==========
 
-- Wheels for Python 3.8
+- Wheels for Python 3.9
```
Comment (mattip) [inline doc/source/release/1.17.3-notes.rst:18]: I don't think this is correct: 1.17 did not release wheels for 3.9, right?

File: doc/source/release/1.22.0-notes.rst [hunk 6]
```diff
@@ -120,7 +120,7 @@ function.
 Python 3.7 is no longer supported
 ---------------------------------
 Python support has been dropped. This is rather strict, there are changes that
-require Python >= 3.8.
+require Python >= 3.9.
```
Comment (mattip) [inline doc/source/release/1.22.0-notes.rst:123]: I don't think this is correct.

File: numpy/f2py/capi_maps.py [hunk 7]
```diff
@@ -56,6 +56,7 @@
             'complex_double': 'complex',
```
Comment (mattip) [inline numpy/f2py/capi_maps.py:56]: Are these changes connected to the issue at hand?

**Comment by soma2000-lang** [comment]
Will do all the required changes by the end of the day.

**Comment by charris** [comment]
I'd be inclined to close this.

File: doc/source/release/1.17.3-notes.rst (hunk 5 above)
Comment (Neilblaze) [reply doc/source/release/1.17.3-notes.rst:18]: Correct, 3.9 is still up in the air. For 1.17, 3.8 is the latest.

**Comment by mattip** [comment]
Closing. Please try again taking the comments into account.

**Pull request closed without merging**